test/e2e/virtual/syncer/storage_pv_test.go

```diff
@@ -0,0 +1,291 @@
+	}{
+		{
+			name: "process upstream PV",
+			work: func(t *testing.T, syncer *framework.StartedSyncerFixture, wsPath logicalcluster.Path, wsClusterName logicalcluster.Name) {
+				ctx, cancelFunc := context.WithCancel(context.Background())
+				t.Cleanup(cancelFunc)
+
+				ns := &corev1.Namespace{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "test-ns",
+					},
+				}
+				syncTargetKey := syncer.ToSyncTargetKey()
+
+				logWithTimestampf(t, "Creating upstream test namespace %s...", ns.Name)
+				_, err = kubeClusterClient.CoreV1().Cluster(wsPath).Namespaces().Create(ctx, ns, metav1.CreateOptions{})
+				require.NoError(t, err)
+
+				// Create Downstream PVC with a delayed status syncing annotation
+
+				pvcNamespaceLocator := shared.NewNamespaceLocator(wsClusterName, syncer.SyncTargetClusterName, types.UID(syncer.SyncerConfig.SyncTargetUID), syncer.SyncerConfig.SyncTargetName, ns.Name)
+				downstreamNamespaceName, err := shared.PhysicalClusterNamespaceName(pvcNamespaceLocator)
+				require.NoError(t, err)
+				pvcNamespaceLocatorAnnotation, err := json.Marshal(pvcNamespaceLocator)
+				require.NoError(t, err)
+
+				_, err = syncer.DownstreamKubeClient.CoreV1().Namespaces().Create(ctx, &corev1.Namespace{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: downstreamNamespaceName,
+						Annotations: map[string]string{
+							shared.NamespaceLocatorAnnotation: string(pvcNamespaceLocatorAnnotation),
+						},
+					},
+				}, metav1.CreateOptions{})
+				require.NoError(t, err)
+
+				downstreamPVC := &corev1.PersistentVolumeClaim{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-pvc",
+						Namespace: downstreamNamespaceName,
+						Labels: map[string]string{
+							workloadv1alpha1.InternalDownstreamClusterLabel:                  syncTargetKey,
+							workloadv1alpha1.ClusterResourceStateLabelPrefix + syncTargetKey: "Sync",
+						},
+						Annotations: map[string]string{
+							shared.NamespaceLocatorAnnotation: string(pvcNamespaceLocatorAnnotation),
+							storage.DelayStatusSyncing:        "true",
+						},
+					},
+					Spec: corev1.PersistentVolumeClaimSpec{
+						AccessModes: []corev1.PersistentVolumeAccessMode{
+							corev1.ReadWriteOnce,
+						},
+					},
+				}
+
+				downstreamPVC, err = syncer.DownstreamKubeClient.CoreV1().PersistentVolumeClaims(downstreamNamespaceName).Create(ctx, downstreamPVC, metav1.CreateOptions{})
+				require.NoError(t, err)
+
+				// Create DownstreamPV linked to the downstream PVC
+				upstreamKcpClient, err := kcpclientset.NewForConfig(syncer.SyncerConfig.UpstreamConfig)
+				require.NoError(t, err)
+
+				syncTarget, err := upstreamKcpClient.Cluster(wsPath).WorkloadV1alpha1().SyncTargets().Get(ctx,
+					syncer.SyncerConfig.SyncTargetName,
+					metav1.GetOptions{},
+				)
+				require.NoError(t, err)
+				pvNamespaceLocator := shared.NewNamespaceLocator(wsClusterName, logicalcluster.From(syncTarget),
```

Comment:
strange that you need to take the logical cluster and UID from the syncTarget, though you can get them from the SyncerConfig, as you did above in line [79](https://github.com/kcp-dev/kcp/pull/2338/files#diff-aeaf51b0a6e9e0efee0df9bfca87018758b76a8eae5eca1da7695c7ce2949c9dR79).

Would be better if the method wold be consistent.